From 5ec2fb163845723f8db407476e190cbeb1d10c46 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Wed, 3 Aug 2016 17:30:13 +0000
Subject: [PATCH] FWAPI-246 fwadm(1M) should use the REQ_ID environment
 variable when present Reviewed by: Alex Wilson <alex.wilson@joyent.com>

---
 src/fw/lib/util/log.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fw/lib/util/log.js b/src/fw/lib/util/log.js
index cad4f7fa..f652f59e 100644
--- a/src/fw/lib/util/log.js
+++ b/src/fw/lib/util/log.js
@@ -328,7 +328,7 @@ function createLogger(opts, readOnly) {
 
     log = bunyan.createLogger({
         name: logName,
-        req_id: opts.req_id || mod_uuid.v4(),
+        req_id: opts.req_id || process.env.REQ_ID || mod_uuid.v4(),
         serializers: serializers,
         streams: streams
     });
-- 
2.21.0

